src/main/java/com/db/crud/voting/service/impl/AgendaServiceImpl.java

```diff
@@ -1,76 +1,86 @@
-package com.db.crud.voting.service.agenda;
+package com.db.crud.voting.service.impl;
 
 import java.util.List;
 import java.util.Optional;
 import java.time.LocalDateTime;
 import java.time.temporal.ChronoUnit;
 
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
-import com.db.crud.voting.dto.mapper.AgendaMapper;
-import com.db.crud.voting.dto.mapper.LogMapper;
-import com.db.crud.voting.dto.mapper.VoteMapper;
 import com.db.crud.voting.dto.request.AddVoteRequest;
 import com.db.crud.voting.dto.request.AgendaRequest;
 import com.db.crud.voting.dto.request.LogObj;
 import com.db.crud.voting.dto.response.AddVoteResponse;
 import com.db.crud.voting.dto.response.AgendaResponse;
 import com.db.crud.voting.enums.Operation;
 import com.db.crud.voting.enums.UserType;
+import com.db.crud.voting.enums.Vote;
 import com.db.crud.voting.exception.AgendaEndedException;
 import com.db.crud.voting.exception.AuthorizationException;
 import com.db.crud.voting.exception.CannotFindEntityException;
 import com.db.crud.voting.exception.EntityExistsException;
 import com.db.crud.voting.exception.UserAlreadyVotedException;
 import com.db.crud.voting.exception.VoteConflictException;
+import com.db.crud.voting.mapper.AgendaMapper;
+import com.db.crud.voting.mapper.LogMapper;
+import com.db.crud.voting.mapper.VoteMapper;
 import com.db.crud.voting.model.Agenda;
 import com.db.crud.voting.model.User;
 import com.db.crud.voting.repository.AgendaRepository;
-import com.db.crud.voting.repository.UserRepository;
-import com.db.crud.voting.service.logs.LogService;
+import com.db.crud.voting.service.AgendaService;
+import com.db.crud.voting.service.LogService;
+import com.db.crud.voting.service.UserService;
 
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 @Service
+@Configuration
 @AllArgsConstructor
+@EnableScheduling
 public class AgendaServiceImpl implements AgendaService {
     
     AgendaRepository agendaRepository;
-    UserRepository userRepository;
+    UserService userService;
     LogService logService;
     AgendaMapper agendaMapper;
     VoteMapper voteMapper;
     LogMapper logMapper;
     
-    @Override
-    public String finishAgenda() {
+    @Scheduled(fixedDelay = 120000)
+    public void finishAgenda() {
         agendaRepository.findByHasEnded(false).forEach(agenda -> {
+            log.debug("Verify if any agendas has finished...");
```

Comment:
```suggestion
            log.debug("Verify if any agendas has finished...");
```

Sugestão: busque saber a **finalidade** e **quando usar** `log.debug`, `log.info` e `log.error`

Reply:
Estudei um pouco sobre o log do @Slf4j e entendi um pouco sobre os niveis

- Info - Informar sobre pontos os quais o código está passando
- Debug - Informar sobre valores de determinadas váriaveis
- Error - Avisar sobre Exceptions não fatais para a aplicação

Com base nisso, mudei os loggers das Services e da ErrorHandler, mais especificamente nesses dois commits: [1](https://github.com/AurorinhaBoreal/VotacaoCRUD-BackEnd/pull/37/commits/0e75ceb302c5a1c7e13ad71b5d4c995d03f50ab3) e [2](https://github.com/AurorinhaBoreal/VotacaoCRUD-BackEnd/pull/37/commits/914582b278e5ee15e6800c5f66639b33312d4a2f).